tests/integration/auth_adapter/test_totp.py

```diff
@@ -53,18 +52,18 @@
 
 
 def get_valid_totp_code(
-    secret: str, when: Optional[datetime] = None, offset: int = 0
+    secret: str, when: datetime | None = None, offset: int = 0
 ) -> str:
     """Generate a valid TOTP code for the given secret."""
     if not when:
-        when = cast(datetime, now_as_utc())
-    return pyotp.TOTP(secret).at(when, offset)
+        when = now_as_utc()
+    return pyotp.TOTP(secret).at(when or now_as_utc(), offset)
```

Comment:
why do you need 'or now_as_utc()' here?

Reply:
that was a leftover from fixing typing issues, removed it again